### PR TITLE
Reduce tracing

### PR DESCRIPTION
**Pull request by SergeyTeplyakov**
This PR reduces the number of messages sent to Kusto by doing the following things:

1. Stop tracing "starts" in `FileSystemContentStoreInternal` (should cut about 10% of overall messages because FSCS is responsible for roughly 30% of all the messages.
2. More efficient traces for `RegisterLocalLocationAsync`. Originally RRLA messages are responsible for 20-25% of all the messages and now there is only 1 message instead of 5.
3. Using short hashes as much as possible. This will reduce the size of the messages and will make investigations slightly easier.
4. Cut redundant messages for copying files.

### CODE REVIEW

File: Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using System.Threading.Tasks;
 using BuildXL.Cache.ContentStore.Distributed.NuCache.InMemory;
 using BuildXL.Cache.ContentStore.Distributed.Utilities;
+using BuildXL.Cache.ContentStore.Hashing;
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs]: I moved ShortHash struct into another namespace that caused some using declaration changes in a few places.

File: Public/Src/Cache/ContentStore/Distributed/Sessions/ReadOnlyDistributedContentSession.cs [hunk 2]
```diff
@@ -473,25 +477,27 @@ public Task PinBulkAsync(Context context, IEnumerable<ContentHash> contentHashes
...
                             }
                             else
                             {
                                 var putResult = await TryCopyAndPutAsync(
                                     OperationContext(context),
                                     contentHashWithSizeAndLocations,
                                     cts,
-                                    urgencyHint);
+                                    urgencyHint,
+                                    // We just traced all the hashes as a result of GetBulk call, no need to trace each and individual hash.
```
Comment (kewilli) [inline Public/Src/Cache/ContentStore/Distributed/Sessions/ReadOnlyDistributedContentSession.cs]: >and [](start = 120, length = 3)

Nit: remove `and`

File: Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs [hunk 3]
```diff
@@ -892,13 +893,14 @@ public async Task<BoolResult> RegisterLocalLocationAsync(OperationContext contex
...
 
+            string extraMessage = string.Empty;
             return await context.PerformOperationAsync(
                 Tracer,
                 async () =>
                 {
-                    var eventContentHashes = new List<ContentHashWithSize>();
-                    var eagerContentHashes = new List<ContentHashWithSize>();
-                    var actions = new List<RegisterAction>();
+                    var eventContentHashes = new List<ContentHashWithSize>(contentHashes.Count);
```
Comment (dannyvv) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs]: if this is perf sensitive, should we use pools?

File: Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs [hunk 4]
```diff
@@ -548,9 +548,9 @@ private Task<HeartbeatResponse> HeartbeatAsync(HeartbeatRequest request, Cancell
                 errorMessage => new HeartbeatResponse { Header = ResponseHeader.Failure(startTime, errorMessage) });
         }
 
-        private void LogRequestHandling([CallerMemberName]string requestType = "Unknown")
+        private void OperationStarted([CallerMemberName]string requestType = "Unknown")
         {
-            _logger.Debug($"{requestType} request handled by GRPC server.");
+            // TODO: Add counter, but don't trace
```
Comment (dannyvv) [inline Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs]: >TODO [](start = 15, length = 4)

is this todo for later? Or intented to do in this pr

File: Public/Src/Cache/ContentStore/Hashing/ContentHash.cs [hunk 5]
```diff
@@ -214,6 +216,14 @@ public override string ToString()
             return Serialize();
         }
 
+        /// <summary>
+        /// Produces shorter string representation of the hash.
+        /// </summary>
+        public string ToShortString()
+        {
+            return new ShortHash(this).ToString();
```
Comment (kewilli) [inline Public/Src/Cache/ContentStore/Hashing/ContentHash.cs]: >new ShortHash(this) [](start = 19, length = 19)

It looks like we'll call this many times. Does it make sense to store `Lazy<ShortHash>` on the ContentHash?

File: Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs (hunk 4 above)
Comment (SergeyTeplyakov) [reply Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs]: This todo is for later. But its not that hard to do. So I'll do it.

File: Public/Src/Cache/ContentStore/Hashing/ContentHash.cs (hunk 5 above)
Comment (SergeyTeplyakov) [reply Public/Src/Cache/ContentStore/Hashing/ContentHash.cs]: new ShortHash(this) won't allocate on the heap, but adding lazy field will increase the size of the ContentHash struct potentially causing perf regression.

**Comment by kewilli** [comment]
>                         _logger.Debug($"Streaming file through GRPC with GZip {(compression == CopyCompression.Gzip ? "on" : "off")}");

Could you also migrate this line into the tracer below? It's okay to lift it to the top of this method.

---
Refers to: Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcContentServer.cs:270 in fd9b55a. [](commit_id = fd9b55a09bcdf9cb8bfb2a73027610eb9725c4eb, deletion_comment = False)

File: Public/Src/Cache/ContentStore/Library/Tracing/ContentSessionTracer.cs [hunk 6]
```diff
@@ -86,7 +86,7 @@ public virtual CounterSet GetCounters()
         public override void GetStatsStart(Context context)
         {
             _getStatsCallCounter.Started();
-            base.GetStatsStart(context);
+            // Don't trace starts to reduce the amount of traces.
```
Comment (kewilli) [inline Public/Src/Cache/ContentStore/Library/Tracing/ContentSessionTracer.cs]: >// [](start = 12, length = 2)

Nit: Move this comment to the top of the method?

**Pull request merged**